security_groups.tf

```diff
@@ -1,31 +1,71 @@
-resource "aws_security_group" "analytical_dataset_generation" {
-  name                   = "analytical_dataset_generation_additional_common"
-  description            = "Contains rules for both EMR cluster master nodes and EMR cluster slave nodes"
+resource "aws_security_group" "adg_master" {
+  name                   = "ADG Master"
+  description            = "Contains rules for ADG master nodes; most rules are injected by EMR, not managed by TF"
   revoke_rules_on_delete = true
   vpc_id                 = data.terraform_remote_state.internal_compute.outputs.vpc.vpc.vpc.id
 }
 
-resource "aws_security_group" "master_sg" {
-  name                   = "analytical_dataset_generation_master_sg"
-  description            = "Contains rules for EMR master"
+resource "aws_security_group" "adg_slave" {
+  name                   = "ADG Slave"
+  description            = "Contains rules for ADG slave nodes; most rules are injected by EMR, not managed by TF"
   revoke_rules_on_delete = true
   vpc_id                 = data.terraform_remote_state.internal_compute.outputs.vpc.vpc.vpc.id
 }
 
-resource "aws_security_group" "slave_sg" {
-  name                   = "analytical_dataset_generation_slave_sg"
-  description            = "Contains rules for EMR slave"
+resource "aws_security_group" "adg_common" {
+  name                   = "ADG Common"
+  description            = "Contains rules for both ADG master and ADG slave nodes"
   revoke_rules_on_delete = true
   vpc_id                 = data.terraform_remote_state.internal_compute.outputs.vpc.vpc.vpc.id
 }
 
-resource "aws_security_group" "service_access_sg" {
-  name                   = "analytical_dataset_generation_service_access_sg"
-  description            = "Contains rules for EMR cluster"
+resource "aws_security_group" "adg_emr_service" {
+  name                   = "ADG EMR Service"
+  description            = "Contains rules for EMR service when managing the ADG cluster; rules are injected by EMR, not managed by TF"
   revoke_rules_on_delete = true
   vpc_id                 = data.terraform_remote_state.internal_compute.outputs.vpc.vpc.vpc.id
 }
 
+resource "aws_security_group_rule" "egress_https_to_vpc_endpoints" {
+  description              = "Allow HTTPS traffic to VPC endpoints"
+  from_port                = 443
+  protocol                 = "tcp"
+  security_group_id        = aws_security_group.adg_common.id
+  to_port                  = 443
+  type                     = "egress"
+  source_security_group_id = data.terraform_remote_state.internal_compute.outputs.vpc.vpc.interface_vpce_sg_id
+}
+
+resource "aws_security_group_rule" "ingress_https_vpc_endpoints_from_emr" {
+  description              = "Allow HTTPS traffic from Analytical Dataset Generator"
+  from_port                = 443
+  protocol                 = "tcp"
+  security_group_id        = data.terraform_remote_state.internal_compute.outputs.vpc.vpc.interface_vpce_sg_id
```

Comment:
What inbound ADG traffic will be initiated from a VPC endpoint?

Reply:
Sorry, read that one the wrong way around. That's the ingress rule for the remote state. As you were :)